src/youtube/js/youtube.js

```diff
@@ -86,6 +86,7 @@ require("../../../html5-common/js/utils/constants.js");
    * @property {object} youtubePlayer The youtube player is creted.
    */
   function onYouTubeIframeAPIReady() {
+    if ( youtubeID == null || youtubePlayer ) { return};
```

Comment:
youtubeID is never null right now. It starts undefined, and only gets set with the setVideoUrl function if youtubeID is truthy. Did you mean to check if youtubeID is falsy, return?